flash/core/trainer.py

```diff
@@ -117,18 +74,36 @@ def finetune(
             val_dataloaders: Either a single Pytorch Dataloader or a list of them, specifying validation samples.
                 If the model has a predefined val_dataloaders method this will be skipped
 
-            unfreeze_milestones: A tuple of two integers. First value marks the epoch in which the last 5
-            layers of the backbone will be unfrozen. The second value marks the epoch in which the full backbone will
-            be unfrozen.
+            finetune_strategy: Should either be a string or a finetuning callback subclassing
+                ``pytorch_lightning.callbacks.BaseFinetuning``.
+
+            callbacks_kwargs: Those arguments will be provided to `model.configure_finetune_callbacks`
+                to instantiante your own finetuning callbacks.
 
         """
-        if hasattr(model, "backbone"):
-            # TODO: if we find a finetuning callback in the trainer should we change it?
-            # or should we warn the user?
-            if not any(isinstance(c, BaseFinetuning) for c in self.callbacks):
-                # TODO: should pass config from arguments
-                self.callbacks.append(MilestonesFinetuningCallback(milestones=unfreeze_milestones))
-        else:
-            warnings.warn("Warning: model does not have a 'backbone' attribute, will train normally")
+        if isinstance(finetune_strategy, Callback) and not isinstance(finetune_strategy, BaseFinetuning):
```

Comment:
BaseFinetuning is a Callback already, isnt it?

```suggestion
        if not isinstance(finetune_strategy, BaseFinetuning):
```